File: myblog-server/src/main/java/com/site/blog/listener/AppListener.java
package com.site.blog.listener;

import org.springframework.boot.web.context.WebServerInitializedEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.context.annotation.Configuration;

import java.net.InetAddress;
import java.net.UnknownHostException;

/**
 * @author yanni
 * @date time 2021/10/29 12:13
 * @modified By:
 */
@Configuration
public class AppListener implements ApplicationListener<WebServerInitializedEvent> {
    @Override
    public void onApplicationEvent(WebServerInitializedEvent event) {
        int port = event.getApplicationContext().getWebServer().getPort();
        InetAddress address = null;
        try {
            address = InetAddress.getLocalHost();
            System.out.println("http://" + address.getHostAddress() + ":" + port + "/swagger-ui.html");
            System.out.println("http://" + address.getHostAddress() + ":" + port);
        } catch (UnknownHostException e) {
            e.printStackTrace();
        }
    }
}

File: blog-sever/server-jpa/src/main/kotlin/com/site/blog/config/listener/AppListener.kt
package com.site.blog.config.listener

import org.springframework.boot.web.context.WebServerInitializedEvent
import org.springframework.context.ApplicationListener
import org.springframework.context.annotation.Configuration
import java.net.InetAddress
import java.net.UnknownHostException

/**
 * @author yanni
 * @date time 2021/10/29 12:13
 * @modified By:
 */
@Configuration
class AppListener : ApplicationListener<WebServerInitializedEvent> {
    override fun onApplicationEvent(event: WebServerInitializedEvent) {
        val port = event.applicationContext.webServer.port
        var address: InetAddress? = null
        try {
            address = InetAddress.getLocalHost()
            println("http://" + address.hostAddress + ":" + port + "/swagger-ui.html")
            println("http://" + address.hostAddress + ":" + port)
        } catch (e: UnknownHostException) {
            e.printStackTrace()
        }
    }
}